File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/li_cwstring_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

li_cwstring

if (count("ab\0ab\0ab\0", 0) != 3)
    error    
endif

if (!strcmp(test1(),"Hello World"))
    error
endif

if (!strcmp(test2()," !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_"))
    error
endif

if (!strcmp(test3("hello"),"hello-suffix"))
    error
endif

if (!strcmp(test4("hello"),"hello-suffix"))
    error
endif
    
if (!strcmp(test5(4),'xxxx'))
    error
endif

if (!strcmp(test6(10),'xxxxx'))
    error
endif
    
if (!strcmp(test7(),"Hello world!"))
    error
endif

if (!strcmp(test8()," !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_"))
    error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/li_cwstring_runme.py
from li_cwstring import *

if count(u"ab\0ab\0ab\0", 0) != 3:
    raise RuntimeError

if test1() != u"Hello World":
    raise RuntimeError

if test2() != u" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_":
    raise RuntimeError

if test3("hello") != u"hello-suffix":
    raise RuntimeError

if test4("hello") != u"hello-suffix":
    raise RuntimeError

if test5(4) != u'xxxx':
    raise RuntimeError

if test6(10) != u'xxxxx':
    raise RuntimeError

if test7() != u"Hello world!":
    raise RuntimeError

if test8() != u" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_":
    raise RuntimeError
